File: frontend/react/src/types/generationJob.ts
// Generic AI generation job (Priority 1). One shape covers every
// generation flow (Sales Playbook, Meeting Brief, Outreach Draft,
// Report) - see backend/schemas/generation_job.py.
export type GenerationJobStatus = "pending" | "running" | "completed" | "failed";

export interface GenerationJob {
  id: string;
  job_type: string;
  status: GenerationJobStatus;
  company_id: string;
  result_id: string | null;
  error_message: string | null;
  progress_message: string | null;
  retry_count: number;
  created_at: string;
  updated_at: string;
}

File: backend/schemas/generation_job.py
"""Response schema for the generic generation job endpoints (Priority 1)."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class GenerationJobOut(BaseModel):
    id: str
    job_type: str
    status: str
    company_id: str
    result_id: Optional[str] = None
    error_message: Optional[str] = None
    progress_message: Optional[str] = None
    retry_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
